Core/GDCore/Project/Variable.h

```diff
@@ -28,15 +28,15 @@ namespace gd {
  */
 class GD_CORE_API Variable {
  public:
-  enum class Type {
+  enum class Type : char {
     // Primitive types
-    String,
-    Number,
-    Boolean,
+    String = 't',

```

Comment:
This might be a bit cryptic when reading just the JSON. What do you think of serializing the full name ("string", "number", "boolean", etc...)?

Reply:
Enums only accept numbers. I used chars as they are numbers but can represent text. I believe that 't', 'n' etc are more readable than 0, 1, 2 etc. I originally tried to map the enums to string names but making a switch function like 
```cpp
switch (type) {
  case(Type::String): 
    return "string";
  // etc
}
```
didn't work because I cannot return temporary strings. I also tried to make a static map like 
```cpp
static const std::map<Type, gd::String&> TypeNames = {
  {Type::String = "string"}
};
```
Also throwed compilation errors. I feel really stupid but I don't know how I am meant to do it :/

Reply:
I think the simplest is to go for a switch (or a plain old if/else). It's what I've done in EventsFunction.cpp
Not the "prettiest", but if it's just for the serialization and unserialization, that's fair enough (sure you have to remember it if we had one day a new type, but that's a a one in a year or one in a decade event :)) 

If you really want to have a separate function (which would be a good idea in case you need this in more than one place), you can make one returning a gd::String directly (not a const reference): 
`gd::String TypeToString(gd::Variable::Type type) { if (type == Type::String) return "string"; else if(...)... }`

(you put this function either in the .cpp file directly, inside the namespace, or as a private, static function of the class, whatever you prefer).

It will work because you don't return a const ref, you're returning a newly constructed gd::String (and if you're concerned about unecessary generation of strings, there will be the "return value optimization" applied anyway by the C++ compiler: https://en.wikipedia.org/wiki/Copy_elision).

If it's just for serialization, let's go with a few if/else :)